packages/core/test/smoke.test.ts

```diff
@@ -369,3 +379,15 @@ describe("shared can be accessed on both server and client", () => {
     globalThis.window = window;
   });
 });
+
+test("envs are readonly", () => {
+  const env = createEnv({
+    server: { BAR: z.string() },
+    runtimeEnv: { BAR: "bar" },
+  });
+
+  // @ts-expect-error - envs are readonly
+  env.BAR = "foo";
+
+  expect(env).toMatchObject({ BAR: "bar" });
```

Comment:
this currently fails

Reply:
the test failing is intentional because TypeScript doesn't exist at runtime

Reply:
i'm not sure what the point of this test is.
the other tests that i modified already ensure that the type is readonly, so i think we can delete this test.

Reply:
fixed it by making the `set` handler throw

Reply:
i'm against this test and this `set` handler for the reasons i outlined in my previous post